[Dart] Remove content type from header when content type is not specified

gasugesu opened the pull request:
<!-- Enter details of the change here. Include additional tests that have been done, reference to the issue for tracking, etc. -->

The current `Dart Client` does not require `ContentType` to specify `Content-type: application/json`, which is incorrect.  
There are two reasons.  
  
1. Specifying `Content-type: application/json` in `header` without declaration can be confusing.
2. There may be no RequestBody at all. In such case, I don't think `Content-type: application/json` is correct.
  
Therefore, if `ContentType` is not specified, `header` should not add `Content-type: application/json`  

<!-- Please check the completed items below -->
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@ircecho (2017/07) @swipesight (2018/09) @jaumard (2018/09) @nickmeinhold (2019/09) @athornz (2019/12) @amondnet (2019/12)

wing328 commented:
CI failed with the following errors:
```
Precompiled test:test.
00:03 +0 -1: test/store_faked_client_test.dart: Store API with faked client places an order and gets it by id [E]                                                                                      
  Bad state: POST was called with unexpected headers: {Content-Type: application/json} should be null
  test/fake_client.dart 49:7                 FakeClient.post
  package:openapi/api_client.dart 137:25     ApiClient.invokeAPI
  package:openapi/api/store_api.dart 214:36  StoreApi.placeOrderWithHttpInfo
  package:openapi/api/store_api.dart 229:31  StoreApi.placeOrder
  test/store_faked_client_test.dart 33:22    main.<fn>.<fn>
  
00:03 +0 -2: test/store_faked_client_test.dart: Store API with faked client deletes an order [E]                                                                                                       
  Bad state: POST was called with unexpected headers: {Content-Type: application/json} should be null
  test/fake_client.dart 49:7                 FakeClient.post
  package:openapi/api_client.dart 137:25     ApiClient.invokeAPI
  package:openapi/api/store_api.dart 214:36  StoreApi.placeOrderWithHttpInfo
  package:openapi/api/store_api.dart 229:31  StoreApi.placeOrder
  test/store_faked_client_test.dart 54:22    main.<fn>.<fn>
  
00:03 +1 -2: test/user_test.dart: User API with live client creates a user                                                                                                                                            
  Skip: e2e tests for CI
00:03 +1 ~1 -2: test/user_test.dart: User API with live client creates users with list input                                                                                                                          
  Skip: e2e tests for CI
00:03 +1 ~2 -2: test/user_test.dart: User API with live client updates a user                                                                                                                                         
  Skip: e2e tests for CI
00:03 +1 ~3 -2: test/user_test.dart: User API with live client deletes a user                                                                                                                                         
  Skip: e2e tests for CI
00:03 +1 ~4 -2: test/user_test.dart: User API with live client logs a user in                                                                                                                                         
  Skip: e2e tests for CI
00:05 +1 ~5 -2: test/pet_test.dart: Pet API with live client adds a new pet and gets it by id                                                                                                                         
  Skip: e2e tests for CI
00:05 +1 ~6 -2: test/pet_test.dart: Pet API with live client doesn't get non-existing pet by id                                                                                                                       
  Skip: e2e tests for CI
00:05 +1 ~7 -2: test/pet_test.dart: Pet API with live client deletes existing pet by id                                                                                                                               
  Skip: e2e tests for CI
00:05 +1 ~8 -2: test/pet_test.dart: Pet API with live client updates pet with form                                                                                                                                    
  Skip: e2e tests for CI
00:05 +1 ~9 -2: test/pet_test.dart: Pet API with live client updates existing pet                                                                                                                                     
  Skip: e2e tests for CI
00:05 +1 ~10 -2: test/pet_test.dart: Pet API with live client finds pets by status                                                                                                                                    
  Skip: e2e tests for CI
00:05 +1 ~11 -2: test/pet_test.dart: Pet API with live client uploads a pet image                                                                                                                                     
  Skip: e2e tests for CI
00:08 +16 ~12 -2: Some tests failed.                                                                                                                                                                             
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2.1:exec (pub-test) on project Dart2PetstoreClientTests: Command execution failed. Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
```

I wonder if you can take a look when you've time.

gasugesu commented:
I have fixed tests.
The `consumes` type of  'placeOrder' operation is specified as application/json in `modules/openapi-generator/src/test/resources/2_0/petstore.yaml`.
So I have changed `expectedHeaders` from null to application/json and fixed issues.

```
  /store/order:
    post:
      tags:
        - store
      summary: Place an order for a pet
      description: ''
      operationId: placeOrder
      consumes:
        - application/json
        - application/xml
      produces:
        - application/json
        - application/xml
      parameters:
        - in: body
          name: body
          description: order placed for purchasing the pet
          required: true
          schema:
            $ref: '#/definitions/Order'
```

@wing328 Please review agiain.

wing328 commented:
@gasugesu thanks for the PR, which has been included in the v4.3.1 release: https://twitter.com/oas_generator/status/1258057660974329861
